pkg/lightning/backend/external/writer.go

```diff
@@ -531,10 +514,14 @@ func (w *Writer) flushKVs(ctx context.Context, fromClose bool) (err error) {
 		zap.Int("sequence-number", w.currentSeq),
 	)
 	sortStart := time.Now()
-	var dupFound bool
+	var (
+		dupFound bool
+		dupLoc   atomic.Pointer[membuf.SliceLocation]
+	)
+	dupLoc.Store(nil)
```

Comment:
it is nil initially, and we are not sorting concurrently, i think it's ok to use a simple pointer